ibis/backends/impala/tests/snapshots/test_exprs/test_filter_with_analytic/out.sql

```diff
@@ -1,15 +1,18 @@
-SELECT t0.`col`, t0.`analytic`
+WITH t0 AS (
+  SELECT t4.`col`, NULL AS `filter`
+  FROM x t4
+),
+t1 AS (
+  SELECT t0.*
+  FROM t0
+  WHERE t0.`filter` IS NULL
+),
+t2 AS (
+  SELECT t1.`col`, t1.`filter`
+  FROM t1
+)
+SELECT t3.`col`, t3.`analytic`
```

Comment:
We could try to eliminate unnecessary selects in the future.

Reply:
Yep!